app/src/main/java/org/mozilla/fenix/home/collections/TabInCollection.kt

```diff
@@ -0,0 +1,300 @@
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Constraints
+import androidx.compose.ui.unit.dp
+import mozilla.components.browser.state.state.recover.RecoverableTab
+import mozilla.components.concept.engine.Engine
+import mozilla.components.feature.tab.collections.Tab
+import org.mozilla.fenix.R.drawable
+import org.mozilla.fenix.R.string
+import org.mozilla.fenix.compose.inComposePreview
+import org.mozilla.fenix.compose.list.FaviconListItem
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.ext.toShortUrl
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.theme.Theme
+
+/**
+ * Rectangular shape with only right angles used to display a middle tab.
+ */
+private val middleTabShape = RoundedCornerShape(0.dp)
+
+/**
+ * Rectangular shape with only the bottom corners rounded used to display the last tab in a collection.
+ */
+private val bottomTabShape = RoundedCornerShape(bottomStart = 8.dp, bottomEnd = 8.dp)
+
+/**
+ * Display an individual [Tab] as part of a collection.
+ *
+ * @param tab [Tab] to display.
+ * @param isLastInCollection Whether the tab is to be shown between others or as the last one in collection.
+ * @param onClick Invoked when the user click on the tab.
+ * @param onRemove Invoked when the user removes the tab informing also if the tab was swiped to be removed.
+ */
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun TabInCollection(
+    tab: Tab,
+    isLastInCollection: Boolean,
+    onClick: () -> Unit,
+    onRemove: (Boolean) -> Unit,
+) {
+    val dismissState = rememberDismissState()
+
+    if (dismissState.isDismissed(StartToEnd) || dismissState.isDismissed(EndToStart)) {
+        onRemove(true)
+    }
+
+    SwipeToDismiss(
+        state = dismissState,
+        background = {
+            DismissedTabBackground(
+                dismissDirection = dismissState.dismissDirection,
+                isLastInCollection = isLastInCollection,
+            )
+        }
+    ) {
+        // We need to clip the top bounds to avoid this item drawing shadows over the above item.
+        // But we need to add this shadows back to have a clearer separation between tabs
+        // when one is being swiped away.
+        val clippingModifier by remember {
+            derivedStateOf {
+                try {
+                    if (dismissState.progress.fraction != 1f) Modifier else Modifier.clipTop()
+                } catch (e: NoSuchElementException) {
+                    // `androidx.compose.material.Swipeable.findBounds` couldn't find anchors.
+                    // Happened once in testing when deleting a tab. Could not reproduce afterwards.
+                    Modifier.clipTop()
+                }
+            }
+        }
+
+        Card(
+            modifier = clippingModifier
+                .fillMaxWidth()
+                .defaultMinSize(minHeight = 56.dp),
```

Comment:
`defaultMinSize` may not be necessary, since `FaviconListItem` has this already set.